zipkin-lens/src/components/TracePage/TraceSummary.jsx

```diff
@@ -80,19 +83,34 @@ const TraceSummary = React.memo(({ traceSummary }) => {
   }, [isRootedTrace, rootSpanIndex, traceSummary.spans]);
 
   const shownTree = useMemo(() => {
-    const allHiddenSpanIds = {};
-    rerootedTree.forEach((span) => {
-      if (childrenHiddenSpanIds[span.parentId]) {
-        allHiddenSpanIds[span.spanId] = true;
+    let depth = 0;
+    let skip = false;
+
+    return rerootedTree.reduce((acc, cur) => {
```

Comment:
Let's use semantic names instead of `acc`, `cur` - I think it's `shownTraces`, `trace`?